### PR TITLE
CCITCARBON-145 Allow jinja templating within teflo.cfg

### PR DESCRIPTION
**Pull request by rujutashinde**
* rectified task_concurrency method in helpers.py
* Added unit test to cover this change
* Added doc change in configuration.rst

### CODE REVIEW

**Review by JunqiZhang0 — CHANGES_REQUESTED** [review]

File: docs/users/configuration.rst [hunk 1]
```diff
@@ -30,6 +30,40 @@ Configuration example (with all options):
...
+Teflo uses Jinja2 template engine to be able to template variables
+within the teflo.cfg file. Teflo allows template variable data to be
+set as environmental variables
+
+Here is an example teflo.cfg file using Jinja to template some variable data:
+
+.. code-block:: bash
+
+    [credentials:openstack]
+    auth_url=https://rhos-d.infra.prod.upshift.rdu2.redhat.com:13000
```
[the remark this answers is not in the document]
Comment (rujutashinde) [reply docs/users/configuration.rst:45]: yup i will remove this and just put <auth_url> instead. good catch!

[the remark this answers is not in the document]
Comment (rujutashinde) [reply docs/users/configuration.rst]: done

File: docs/users/configuration.rst [hunk 2]
```diff
@@ -30,6 +30,40 @@ Configuration example (with all options):
...
+    report=False
+    orchestrate={{ ORC_TASK_CONCURRENCY }}
+
+Prior to running teflo, the templated variables will have to be exported
+
+.. code-block:: bash
+
+    export OS_USER=user1
+    export OS_PASSWORD=password
+    export OS_TENTAT=project1
```
Comment (shay6) [inline docs/users/configuration.rst:62]: s/TENTAT/TENANT

**Review by shay6 — APPROVED** [review]

File: docs/users/configuration.rst (hunk 2 above)
Comment (rujutashinde) [reply docs/users/configuration.rst]: done

**Review by JunqiZhang0 — APPROVED** [review]

**Pull request merged**